src/Benchmark/FormatterBenchmark.cs

```diff
@@ -0,0 +1,54 @@
+using BenchmarkDotNet.Attributes;
+using JustEat.StatsD;
+
+namespace Benchmark
+{
+    [MemoryDiagnoser]
+    public class FormatterBenchmark
+    {
+        private StatsDMessageFormatter _formatter;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _formatter = new StatsDMessageFormatter("test.prefix");
+        }
+
+        [Benchmark]
+        public string IncrementBy1()
+        {
+            return _formatter.Increment(1, "some.stat");
```

Comment:
Could all these methods just be `void` given that BenchmarkDotNet throws away the return values anyway?

Reply:
IIRC, BenchmarkDotNet  docs say to return a value so that code won't be optimised out.

Reply:
I guess it depends on the case? The compiler couldn’t optimise everything away in case there was a side effect. Always returning probably removes the need to worry about whether it will or not.

Reply:
I read it here http://benchmarkdotnet.org/Guides/GoodPractices.htm